### PR TITLE
Fix repr(Rust) layout

### LINKED ISSUE #4: Enforce repr(C)?
The crate today doesn't really work on non-repr(C) types. We could enforce only C structs, or we could sort fields by offset before running layout calculation.

### PR DESCRIPTION
**Pull request by Gilnaa**
- Added `offset` as a member of `Field`
 - Sort by `offset`
 - Paddings are inserted on format, now in the correct places.

`cargo-run` in `try-crate` now prints
```
Foo (size 4, alignment 2)
| Offset | Name      | Size |
| ------ | --------- | ---- |
| 0      | a         | 1    |
| 1      | [padding] | 1    |
| 2      | b         | 2    |

Bar (size 4, alignment 2)
| Offset | Name      | Size |
| ------ | --------- | ---- |
| 0      | b         | 2    |
| 2      | a         | 1    |
| 3      | [padding] | 1    |
```

It's a bit of a shift from the original, not sure if that's okay or not.

### CODE REVIEW

**Comment by LPGhatguy** [comment]
This looks great to me! I like that padding becomes implicitly encoded by the offset member of each field. It's a lot more clear to diagnose issues with that format.

I've approved the CI run, and once it's green I'll merge this!

**Comment by Gilnaa** [comment]
Sorry, forgot to run `cargo fmt`, should be fixed now :)

**Review by LPGhatguy — APPROVED** [review]
Awesome, thank you!

**Pull request merged**
